backend/src/rss.rs

```diff
@@ -0,0 +1,204 @@
+            <channel>
+                <title>{series_title}</title>
+                <link>{series_link}</link>
+                <description>{series_description}</description>
+                <language>"und"</language>
+                <itunes:category text="Education">
+                    <itunes:category text="Courses"/>
+                </itunes:category>
+                <itunes:explicit>"true"</itunes:explicit>
+                <itunes:image href={format!("{tobira_url}/~assets/logo-small.svg")} />
+                <atom:link href={rss_link} rel="self" type="application/rss+xml" />
+                {|buf| video_items(buf, &db, &series_oc_id, &series_title, &rss_link, tobira_url).await.unwrap()}
+            </channel>
+        </rss>
+    );
+
+    Ok(buf.into_string())
+}
+
+/// Generates the single video items of a series in Tobira for inclusion in an RSS feed.
+async fn video_items(
+    doc: &mut ogrim::Document,
+    db: &Client,
+    series_oc_id: &str,
+    series_title: &str,
+    rss_link: &str,
+    tobira_url: &HttpHost,
+) -> Result<(), Error> {
+    let selection = Event::select();
+    let query = format!("select {selection} from events where part_of = $1");
+    let rows = db.query_raw(&query, dbargs![&series_oc_id]).await?;
+
+    rows.try_for_each(|row| {
+        let event = Event::from_row_start(&row);
+        
+        let mut buf = [0; 11];
+        let tobira_event_id = event.id.to_base64(&mut buf);
+        let event_link = format!("{tobira_url}/!v/{tobira_event_id}");
+        let thumbnail = &event.thumbnail_url.unwrap_or_default();
+        let tracks = preferred_tracks(event.tracks);
+        let enclosure_track = tracks.1.unwrap();
+        let enclosure_url = &enclosure_track.uri;
+        let mimetype = &enclosure_track.mimetype.unwrap();
+
+        xml!(doc,
+            <item>
+                <title>{event.title}</title>
+                <link>{event_link}</link>
+                <description>{event.description.unwrap_or_default()}</description>
+                <dc:creator>{event.creators.join(", ")}</dc:creator>
+                <pubDate>{event.created.to_rfc2822()}</pubDate>
+                <guid>{event_link}</guid>
+                <media:thumbnail url={thumbnail} />
+                <itunes:image href={thumbnail} />
+                <enclosure
+                    url={enclosure_url}
+                    type={mimetype}
+                    length="0"
+                />
+                <source url={rss_link}>{series_title}</source>
+                <media:group>
+                    {|doc| for track in tracks.0 {
+                        xml!(doc,
+                            <media:content
+                                url={track.uri}
+                                type={track.mimetype.unwrap()}
+                                medium={"video"}
+                                height={track.resolution.unwrap()[1]}
+                                width={track.resolution.unwrap()[0]}
```

Comment:
The `unwrap`s are inappropriate here. If a track has no given resolution then this currently panics, resulting in a 500. That's not appropriate. There are actually audio-only tracks for example. And in that case `medium` is also not `"video"`. Similar with `mimetype`: it shouldn't panic. All of these attributes are optional in the RSS standard, so we can just omit them if we don't have a default value.

Well "just"... that's currently slightly awkard with `ogrim`. Sigh. Let me think about that.

Reply:
Hm yeah. I don't disagree but I also have no idea how I would do that, sooo any suggestion would be appreciated, if you do come up with something.